Update README transpiler example usage

mtreinish opened the pull request:
<!--
⚠️ If you do not respect this template, your pull request will be closed.
⚠️ Your pull request title should be short detailed and understandable for all.
⚠️ Also, please add a release note file using reno if the change needs to be
  documented in the release notes.
⚠️ If your pull request fixes an open issue, please link to the issue.

- [ ] I have added the tests to cover my changes.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the CONTRIBUTING document.
-->

### Summary

This commit updates the README transpiler usage to use better practices slightly. The first is it changes the use of loose constraints to create a Target explicitly. While the transpile() function does this internally it's good to get people thinking in terms of the target up front so this constructs the target from the same basic values. Secondly it removes the optimization level 3, transpile() has a default optimization level of 2 now and we should just use that as it's the best balance between runtime and optimization. Optimization level 3 spends more runtime and typically doesn't yield significantly better results.


### Details and comments

mtreinish committed:
Update README transpiler example usage

This commit updates the README transpiler usage to use better practices
slightly. The first is it changes the use of loose constraints to create
a Target explicitly. While the transpile() function does this internally
it's good to get people thinking in terms of the target up front so this
constructs the target from the same basic values. Secondly it removes
the optimization level 3, transpile() has a default optimization level
of 2 now and we should just use that as it's the best balance between
runtime and optimization. Optimization level 3 spends more runtime and
typically doesn't yield significantly better results.

jakelishman commented on README.md:
```diff
@@ -97,7 +97,12 @@ The following code will map the example circuit to the `basis_gates = ["cz", "sx
+from qiskit.transpiler import Target, CouplingMap
+target = Target.from_configuration(
+    basis_gates=["cz", "sx", "rz"],
+    coupling_map=CouplingMap.from_line(3, bidirectional=False),
```
If we're using a cz, let's make it bidirectional - it's meaningless to be directed.

mtreinish committed:
Make coupling map bidirectional

The basis gates use cz for the entangling gate which is bidirectional,
so we should use a bidirectional coupling map to reflect this.

mtreinish committed:
Add missing keyword
